tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -2913,19 +2857,14 @@ Status FragmentMetadata::load_tile_offsets(ConstBuffer* buff) {
   // For all attributes, get the tile offsets
   for (unsigned int i = 0; i < attribute_num + 1; ++i) {
     // Get number of tile offsets
-    st = buff->read(&tile_offsets_num, sizeof(uint64_t));
-    if (!st.ok()) {
-      return LOG_STATUS(Status_FragmentMetadataError(
-          "Cannot load fragment metadata; Reading number of tile offsets "
-          "failed"));
-    }
+    tile_offsets_num = deserializer.read<uint64_t>();
 
     if (tile_offsets_num == 0)
       continue;
 
     auto size = tile_offsets_num * sizeof(uint64_t);
     if (memory_tracker_ != nullptr && !memory_tracker_->take_memory(size)) {
-      return LOG_STATUS(Status_FragmentMetadataError(
+      throw StatusException(Status_FragmentMetadataError(
```

Comment:
throw FragmentMetadataStatusException

Reply:
Search in this file to replace all occurence.

Reply:
changed